node/node.go

```diff
@@ -138,27 +138,22 @@ func NewNode(ctx context.Context, opts ...Option) (*Node, error) {
 		return nil, err
 	}
 
-	var node *net.Node
+	var peer *net.Peer
 	if !options.disableP2P {
 		// setup net node
-		node, err = net.NewNode(ctx, db, netOpts...)
+		peer, err = net.NewPeer(ctx, db.Rootstore(), db.Blockstore(), db.Events(), netOpts...)
```

Comment:
thought: since we have access to both the `db` and `peer` here, would it be simpler to just set the `peerID` directory on the `db`?

Something like this:
```go
db.SetPeerID(peer.ID())
```

Another approach that takes a bit more work:
```go
mstore := datstore.NewMultistore(...)
events := event.NewBus()
peer, err := net.NewPeer(ctx, mstore.Rootstore(), mstore.Blockstore(), events)
db, err := db.NewDB(ctx, mstore, events, peer.ID())
```

Reply:
Maybe it makes more sense to do this in the `Start` method instead.

Reply:
I had thought of the first approach and didn't like it. I do like the second approach though. I'll see what I can do.

Reply:
Decided to stay with the current implementation for now as it offers more safety/guarantee that the peer ID being stored on `db` is actually the right one.